cpp/src/rolling/rolling.cu

```diff
@@ -0,0 +1,260 @@
+
+    cudaDeviceSynchronize();
+    CUDA_CHECK_LAST();
+
+    POP_RANGE();
+  }
+
+  /**
+   * @brief If we cannot perform aggregation on this type then throw an error
+   */
+  template<typename ColumnType, template <typename AggType> class agg_op, bool average, class... TArgs,
+	   typename std::enable_if_t<!std::is_arithmetic<ColumnType>::value, std::nullptr_t> = nullptr> 
+  void dispatch_aggregation_type(gdf_size_type nrows, TArgs... FArgs)
+  {
+    CUDF_FAIL("Unsupported column type. Only arithmetic types are supported in rolling windows");
+  }
+
+  /**
+   * @brief Helper function for gdf_rolling. Deduces the type of the
+   * aggregation column and type and calls another function to invoke the
+   * rolling window kernel.
+   */
+  template <typename ColumnType>
+  void operator()(gdf_size_type nrows,
+		  gdf_agg_op agg_type,
+		  void *out_col_data_ptr, gdf_valid_type *out_col_valid_ptr,
+		  void *in_col_data_ptr, gdf_valid_type *in_col_valid_ptr,
+		  gdf_size_type window,
+		  gdf_size_type min_periods,
+		  gdf_size_type forward_window,
+		  const gdf_size_type *window_col,
+		  const gdf_size_type *min_periods_col,
+		  const gdf_size_type *forward_window_col)
+  {
+    switch (agg_type) {
+    case GDF_SUM:
+      dispatch_aggregation_type<ColumnType, sum_op, false>(nrows, reinterpret_cast<ColumnType*>(out_col_data_ptr), out_col_valid_ptr, reinterpret_cast<ColumnType*>(in_col_data_ptr), in_col_valid_ptr, window, min_periods, forward_window, window_col, min_periods_col, forward_window_col);
+      break;
+    case GDF_MIN:
+      dispatch_aggregation_type<ColumnType, min_op, false>(nrows, reinterpret_cast<ColumnType*>(out_col_data_ptr), out_col_valid_ptr, reinterpret_cast<ColumnType*>(in_col_data_ptr), in_col_valid_ptr, window, min_periods, forward_window, window_col, min_periods_col, forward_window_col);
+      break;
+    case GDF_MAX:
+      dispatch_aggregation_type<ColumnType, max_op, false>(nrows, reinterpret_cast<ColumnType*>(out_col_data_ptr), out_col_valid_ptr, reinterpret_cast<ColumnType*>(in_col_data_ptr), in_col_valid_ptr, window, min_periods, forward_window, window_col, min_periods_col, forward_window_col);
+      break;
+    case GDF_COUNT:
+      dispatch_aggregation_type<ColumnType, count_op, false>(nrows, reinterpret_cast<ColumnType*>(out_col_data_ptr), out_col_valid_ptr, reinterpret_cast<ColumnType*>(in_col_data_ptr), in_col_valid_ptr, window, min_periods, forward_window, window_col, min_periods_col, forward_window_col);
+      break;
+    case GDF_AVG:
+      dispatch_aggregation_type<ColumnType, sum_op, true>(nrows, reinterpret_cast<ColumnType*>(out_col_data_ptr), out_col_valid_ptr, reinterpret_cast<ColumnType*>(in_col_data_ptr), in_col_valid_ptr, window, min_periods, forward_window, window_col, min_periods_col, forward_window_col);
+      break;
+    default:
+      // TODO: need a nice way to convert enums to strings, same would be useful for groupbys
```

Comment:
I think we should consolidate our aggregation enums (for reductions, scans, rolling, groupby) and make an `aggregate_dispatcher` that works like `type_dispatcher`. Then we can also solve the `to_string` problem for the enum in that machinery.

Reply:
Agree here. I'm going to update the TODO comment there, but consolidating this will be done outside of this PR.